### PR TITLE
[CDF-511][CDF-512] Reviewed dateFormat function and added dateParse function

### PR DESCRIPTION
**Pull request by davidmsantos90**
Also updated CDF documentation to reflect the changes made

@pamval please review

### CODE REVIEW

**Comment by pamval** [comment]
@davidmsantos90 rebase and make sure this fix applies to legacy and require-js

**Comment by davidmsantos90** [comment]
@pamval rebased and working on both require-js and legacy. please review

**Comment by pamval** [comment]
Where are the unit tests for dateParse  ?

**Comment by davidmsantos90** [comment]
@pamval  Unit Tests for dateParse created. please review

File: cdf-core/cdf/js-modules/dashboard/Utils.js [hunk 1]
```diff
@@ -153,19 +153,51 @@ define(['../Logger', 'amd!../lib/underscore', '../lib/moment', '../lib/CCC/cdo',
...
+   * or the one that the user specified if it exists, otherwise
+   * uses the default language 'en-US'
    *
-   * @mnethod dateFormat
-   * @param mask
+   * @method dateFormat
    * @param date
+   * @param mask
+   * @param langCode
    * @returns {string} formatted date
```
Comment (pamval) [inline cdf-core/cdf/js-modules/dashboard/Utils.js]: Provide a short description for each parameter.

File: cdf-core/cdf/js-modules/dashboard/Utils.js [hunk 2]
```diff
@@ -153,19 +153,51 @@ define(['../Logger', 'amd!../lib/underscore', '../lib/moment', '../lib/CCC/cdo',
...
+
+  /**
+   * Parse a date with a given mask
+   *
+   * @method dateParse
+   * @param date
+   * @param mask
+   * @returns {Date} parsed date as a Date object
+   * @static
+   */
```
Comment (pamval) [inline cdf-core/cdf/js-modules/dashboard/Utils.js]: Provide a short description for each parameter

**Comment by davidmsantos90** [comment]
@pamval documentation updated. please review

**Pull request merged**